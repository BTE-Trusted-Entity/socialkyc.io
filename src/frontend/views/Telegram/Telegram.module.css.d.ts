export const __esModule: true;
export const heading: string;
export const buttonsLine: string;
export const iframe: string;
export const iframeLoading: string;
export const profile: string;
