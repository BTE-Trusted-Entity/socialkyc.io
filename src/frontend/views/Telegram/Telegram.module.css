.heading {
  composes: heading from '../../components/CredentialFlow/CredentialFlow.module.css';
  background-image: url('../Attester/telegram.svg');
}

.buttonsLine {
  text-align: center;
}

.iframe {
  overflow: hidden;
  border: medium none;
  height: 2rem;
}

.iframeLoading {
  composes: iframe;
  visibility: hidden;
}

.profile {
  display: grid;
  grid-template-areas: 'key value' 'key value';
  width: calc(250rem / 16);
  gap: 0.5rem;
  margin-bottom: 2rem;
}
