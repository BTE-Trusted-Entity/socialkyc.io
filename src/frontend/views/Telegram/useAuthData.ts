import { useEffect, useState } from 'react';

export function useAuthData() {
  const [authData, setAuthData] = useState<string>();

  useEffect(() => {
    async function handleMessage({ data, origin }: MessageEvent) {
      try {
        if (
          origin === 'https://oauth.telegram.org' &&
          JSON.parse(data).event === 'auth_user'
        ) {
          setAuthData(data);
        }
      } catch {
        // ignore the errors, may come from unrelated events
      }
    }

    window.addEventListener('message', handleMessage);
    return () => window.removeEventListener('message', handleMessage);
  }, []);

  return authData;
}
