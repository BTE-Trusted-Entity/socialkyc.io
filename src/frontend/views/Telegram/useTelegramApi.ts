import { useMemo } from 'react';

import { confirmTelegram } from '../../../backend/telegram/confirmTelegramApi';
import { authUrlTelegram } from '../../../backend/telegram/authUrlTelegramApi';
import { quote } from '../../../backend/endpoints/quoteApi';
import { requestAttestationTelegram } from '../../../backend/telegram/requestAttestationTelegramApi';
import { attest } from '../../../backend/endpoints/attestApi';

import { bindToSession } from '../../utilities/bindToSession';

export function useTelegramApi(sessionId: string) {
  return useMemo(() => {
    const sessionBound = bindToSession(sessionId);
    return {
      authUrl: sessionBound(authUrlTelegram),
      confirm: sessionBound(confirmTelegram),
      quote: sessionBound(quote.bind(undefined, 'telegram')),
      requestAttestation: sessionBound(requestAttestationTelegram),
      attest: sessionBound(attest),
    };
  }, [sessionId]);
}
