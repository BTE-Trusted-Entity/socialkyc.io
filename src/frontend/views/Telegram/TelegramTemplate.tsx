import {
  FormEventHandler,
  Fragment,
  MouseEventHandler,
  ReactNode,
} from 'react';
import { Prompt } from 'react-router-dom';
import cx from 'classnames';

import * as flowStyles from '../../components/CredentialFlow/CredentialFlow.module.css';
import * as styles from './Telegram.module.css';

import { Spinner } from '../../components/Spinner/Spinner';
import { usePreventNavigation } from '../../utilities/usePreventNavigation';
import { OAuthExplainer } from '../../components/OAuthExplainer/OAuthExplainer';
import { LinkBack } from '../../components/LinkBack/LinkBack';
import { AttestationErrorUnknown } from '../../components/AttestationErrorUnknown/AttestationErrorUnknown';
import { ExpiryDate } from '../../components/ExpiryDate/ExpiryDate';
import { DetailedMessage } from '../../components/DetailedMessage/DetailedMessage';
import { AttestationProcessAnchoring } from '../../components/AttestationProcessAnchoring/AttestationProcessAnchoring';
import { AttestationProcessReady } from '../../components/AttestationProcessReady/AttestationProcessReady';
import { SigningErrorClosed } from '../../components/SigningErrorClosed/SigningErrorClosed';

import { TelegramProfile } from './Telegram';

export type AttestationStatus =
  | 'none'
  | 'urlReady'
  | 'authorizing'
  | 'authorized'
  | 'attesting'
  | 'ready'
  | 'error';

export type FlowError = 'unauthorized' | 'closed' | 'unknown';

interface Props {
  status: AttestationStatus;
  processing: boolean;
  handleSubmit: FormEventHandler;
  handleBackup: MouseEventHandler;
  handleTryAgainClick: MouseEventHandler;
  authUrlLoader?: ReactNode;
  flowError?: FlowError;
  profile?: TelegramProfile;
}

export function TelegramTemplate({
  status,
  processing,
  handleSubmit,
  handleBackup,
  handleTryAgainClick,
  authUrlLoader,
  flowError,
  profile,
}: Props): JSX.Element {
  const preventNavigation = usePreventNavigation(
    processing || ['authorizing', 'authorized', 'attesting'].includes(status),
  );

  return (
    <section
      className={cx(flowStyles.container, {
        [flowStyles.processing]: processing,
      })}
      aria-busy={processing}
    >
      {processing && <Spinner />}

      <h1 className={styles.heading}>Telegram Account Attestation</h1>

      <Prompt
        when={preventNavigation}
        message="The Telegram attestation process has already started. Are you sure you want to leave?"
      />

      <OAuthExplainer service="Telegram" />

      {status === 'none' && (
        <DetailedMessage
          icon="spinner"
          heading="Attestation process:"
          message="Generating link"
          details="Please wait, generating Telegram authorization link."
        />
      )}

      {authUrlLoader && <p className={styles.buttonsLine}>{authUrlLoader}</p>}

      {status === 'authorizing' && (
        <DetailedMessage
          icon="spinner"
          heading="Attestation process:"
          message="Authorizing"
          details="Please wait, accessing Telegram account details."
        />
      )}

      {status === 'authorized' && profile && (
        <form onSubmit={handleSubmit}>
          <dl className={styles.profile}>
            <dt>User-ID:</dt>
            <dd>{profile['User ID']}</dd>

            <dt>First name:</dt>
            <dd>{profile['First name']}</dd>

            {profile['Last name'] && (
              <Fragment>
                <dt>Last name:</dt>
                <dd>{profile['Last name']}</dd>
              </Fragment>
            )}

            {profile.Username && (
              <Fragment>
                <dt>Username:</dt>
                <dd>{profile.Username}</dd>
              </Fragment>
            )}
          </dl>

          <p>
            Validity: one year (<ExpiryDate />)
          </p>

          {flowError === 'closed' && <SigningErrorClosed />}

          <button className={flowStyles.ctaButton}>Continue in Wallet</button>
        </form>
      )}

      {status === 'attesting' && <AttestationProcessAnchoring />}

      {status === 'ready' && (
        <Fragment>
          <AttestationProcessReady />
          <button
            className={flowStyles.ctaButton}
            type="button"
            onClick={handleBackup}
          >
            Show credential in wallet
          </button>
        </Fragment>
      )}

      {flowError === 'unknown' && <AttestationErrorUnknown />}

      {flowError === 'unauthorized' && (
        <DetailedMessage
          icon="exclamation"
          heading="Authorization error:"
          message="There was an error authorizing your Telegram account."
          details="Click “Try Again” or reload the page or restart your browser."
        />
      )}

      {(flowError === 'unknown' || flowError === 'unauthorized') && (
        <button
          type="button"
          className={flowStyles.ctaButton}
          onClick={handleTryAgainClick}
        >
          Try again
        </button>
      )}

      {!preventNavigation && <LinkBack />}
    </section>
  );
}
