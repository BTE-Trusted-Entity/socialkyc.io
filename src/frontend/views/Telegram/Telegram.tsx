import { FormEvent, Fragment, useCallback, useEffect, useState } from 'react';
import { IEncryptedMessage } from '@kiltprotocol/sdk-js';

import * as styles from './Telegram.module.css';

import { Rejection, Session } from '../../utilities/session';
import {
  AttestationStatus,
  FlowError,
  OAuthTemplate,
} from '../../components/OAuthTemplate/OAuthTemplate';
import backgroundImage from '../Attester/telegram.svg';

import { useTelegramApi } from './useTelegramApi';
import { useAuthData } from './useAuthData';

export interface TelegramProfile {
  'User ID': number;
  'First name': string;
  'Last name'?: string;
  Username?: string;
}

interface Props {
  session: Session;
}

export function Telegram({ session }: Props): JSX.Element {
  const [flowError, setFlowError] = useState<FlowError>();
  const [status, setStatus] = useState<AttestationStatus>('none');
  const [processing, setProcessing] = useState(false);

  const telegramApi = useTelegramApi(session.sessionId);

  const [authUrl, setAuthUrl] = useState('');

  const [profile, setProfile] = useState<TelegramProfile>();

  const [backupMessage, setBackupMessage] = useState<IEncryptedMessage>();

  useEffect(() => {
    if (status !== 'none') {
      return;
    }
    (async () => {
      try {
        const url = await telegramApi.authUrl({});
        setAuthUrl(url);
      } catch (error) {
        console.error(error);
        setStatus('error');
        setFlowError('unknown');
      }
    })();
  }, [status, telegramApi]);

  const authData = useAuthData();
  useEffect(() => {
    (async () => {
      if (!authData) {
        return;
      }

      try {
        setStatus('authorizing');
        setProfile(await telegramApi.confirm({ json: authData }));
        setStatus('authorized');
      } catch {
        setStatus('error');
        setFlowError('unauthorized');
      }
    })();
  }, [authData, telegramApi]);

  const handleSubmit = useCallback(
    async (event: FormEvent) => {
      event.preventDefault();
      setProcessing(true);
      setFlowError(undefined);
      try {
        await session.listen(async (message) => {
          try {
            await telegramApi.requestAttestation({ message });
            setStatus('attesting');
            setProcessing(false);

            setBackupMessage(await telegramApi.attest({}));
            setStatus('ready');
          } catch (exception) {
            if (exception instanceof Rejection) {
              setFlowError('closed');
            } else {
              console.error(exception);
              setFlowError('unknown');
              setStatus('error');
            }
          }
        });

        const message = await telegramApi.quote({});

        await session.send(message);
      } catch (error) {
        console.error(error);
        setStatus('error');
        setFlowError('unknown');
      } finally {
        setProcessing(false);
      }
    },
    [telegramApi, session],
  );

  const handleBackup = useCallback(async () => {
    try {
      if (!backupMessage) {
        throw new Error('No backup message');
      }
      await session.send(backupMessage);
    } catch (exception) {
      if (exception instanceof Rejection) {
        return; // donâ€™t care that the user has closed the dialog
      }
      setFlowError('unknown');
      console.error(exception);
    }
  }, [backupMessage, session]);

  const handleTryAgainClick = useCallback(() => {
    setStatus('none');
    setFlowError(undefined);
  }, []);

  const handleLoaded = useCallback(() => setStatus('urlReady'), []);
  const showLoader = ['none', 'urlReady'].includes(status) && authUrl;
  const authUrlLoader = !showLoader ? undefined : (
    <iframe
      src={authUrl}
      className={status === 'urlReady' ? styles.iframe : styles.iframeLoading}
      onLoad={handleLoaded}
    />
  );

  return (
    <OAuthTemplate
      service="Telegram"
      backgroundImage={backgroundImage}
      status={status}
      processing={processing}
      handleSubmit={handleSubmit}
      handleBackup={handleBackup}
      handleTryAgainClick={handleTryAgainClick}
      authUrlLoader={authUrlLoader}
      profile={
        profile && (
          <Fragment>
            <dt>User-ID:</dt>
            <dd>{profile['User ID']}</dd>

            <dt>First name:</dt>
            <dd>{profile['First name']}</dd>

            {profile['Last name'] && (
              <Fragment>
                <dt>Last name:</dt>
                <dd>{profile['Last name']}</dd>
              </Fragment>
            )}

            {profile.Username && (
              <Fragment>
                <dt>Username:</dt>
                <dd>{profile.Username}</dd>
              </Fragment>
            )}
          </Fragment>
        )
      }
      flowError={flowError}
    />
  );
}
