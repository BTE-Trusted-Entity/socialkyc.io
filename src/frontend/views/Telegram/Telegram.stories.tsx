import { Meta } from '@storybook/react';

import { action } from '@storybook/addon-actions';

import { TelegramTemplate } from './TelegramTemplate';
import { TelegramProfile } from './Telegram';

export default {
  title: 'Views/Telegram',
  component: TelegramTemplate,
} as Meta;

const actions = {
  handleSignInClick: action('Sign in'),
  handleSubmit: action('submit'),
  handleBackup: action('backup'),
  handleTryAgainClick: action('tryAgain'),
};

const profileMock: TelegramProfile = {
  'First name': 'TestUser',
  'User ID': 1234556789,
};

export function Start(): JSX.Element {
  return <TelegramTemplate status="none" processing={false} {...actions} />;
}

export function UrlReady(): JSX.Element {
  return <TelegramTemplate status="urlReady" processing={false} {...actions} />;
}

export function Authorizing(): JSX.Element {
  return (
    <TelegramTemplate status="authorizing" processing={false} {...actions} />
  );
}

export function Authorized(): JSX.Element {
  return (
    <TelegramTemplate
      status="authorized"
      processing={false}
      profile={profileMock}
      {...actions}
    />
  );
}

export function QuoteOpen(): JSX.Element {
  return (
    <TelegramTemplate
      status="authorized"
      processing={true}
      profile={profileMock}
      {...actions}
    />
  );
}

export function Attesting(): JSX.Element {
  return (
    <TelegramTemplate status="attesting" processing={false} {...actions} />
  );
}

export function Ready(): JSX.Element {
  return <TelegramTemplate status="ready" processing={false} {...actions} />;
}

export function UnexpectedError(): JSX.Element {
  return (
    <TelegramTemplate
      status="error"
      processing={false}
      flowError="unknown"
      profile={profileMock}
      {...actions}
    />
  );
}

export function Closed(): JSX.Element {
  return (
    <TelegramTemplate
      status="authorized"
      processing={false}
      flowError="closed"
      profile={profileMock}
      {...actions}
    />
  );
}

export function Unauthorized(): JSX.Element {
  return (
    <TelegramTemplate
      status="error"
      processing={false}
      flowError="unauthorized"
      {...actions}
    />
  );
}
