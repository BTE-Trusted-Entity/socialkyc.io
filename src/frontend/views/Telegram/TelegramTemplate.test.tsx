// expect cannot be imported because of https://github.com/testing-library/jest-dom/issues/426
import { describe, it, jest } from '@jest/globals';

import { render } from '../../../testing/testing';

import { TelegramTemplate } from './TelegramTemplate';
import { TelegramProfile } from './Telegram';

jest.useFakeTimers();
jest.setSystemTime(new Date('2022-01-03T12:00:00'));

const actions = {
  handleSignInClick: jest.fn(),
  handleSubmit: jest.fn(),
  handleBackup: jest.fn(),
  handleTryAgainClick: jest.fn(),
};

const profileMock: TelegramProfile = {
  'First name': 'TestUser',
  'User ID': 1234556789,
};

describe('TelegramTemplate', () => {
  it('should match snapshot with status=none', async () => {
    const { container } = render(
      <TelegramTemplate status="none" processing={false} {...actions} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with status=urlReady', async () => {
    const { container } = render(
      <TelegramTemplate
        status="none"
        processing={false}
        {...actions}
        authUrlLoader={<iframe src="/auth-url" className="iframeLoading" />}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with status=urlReady for real', async () => {
    const { container } = render(
      <TelegramTemplate
        status="urlReady"
        processing={false}
        {...actions}
        authUrlLoader={<iframe src="/auth-url" className="iframe" />}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with status=authorizing', async () => {
    const { container } = render(
      <TelegramTemplate status="authorizing" processing={false} {...actions} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with status=authorized', async () => {
    const { container } = render(
      <TelegramTemplate
        status="authorized"
        processing={false}
        profile={profileMock}
        {...actions}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with status=attesting', async () => {
    const { container } = render(
      <TelegramTemplate status="attesting" processing={false} {...actions} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with status=ready', async () => {
    const { container } = render(
      <TelegramTemplate status="ready" processing={false} {...actions} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with flowerror=unauthorized', async () => {
    const { container } = render(
      <TelegramTemplate
        status="error"
        flowError="unauthorized"
        processing={false}
        {...actions}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with flowError=closed', async () => {
    const { container } = render(
      <TelegramTemplate
        status="authorized"
        flowError="closed"
        processing={false}
        profile={profileMock}
        {...actions}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with flowError=unknown', async () => {
    const { container } = render(
      <TelegramTemplate
        status="error"
        flowError="unknown"
        processing={false}
        {...actions}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
