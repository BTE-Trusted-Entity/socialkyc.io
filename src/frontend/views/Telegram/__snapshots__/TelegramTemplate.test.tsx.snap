// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TelegramTemplate should match snapshot with flowError=closed 1`] = `
<div>
  <section
    aria-busy="false"
    class="container"
  >
    <h1
      class="heading"
    >
      Telegram Account Attestation
    </h1>
    <div
      class="explainerContainer"
    >
      <button
        aria-label="Toggle explanation"
        class="showExplainer"
        type="button"
      />
    </div>
    <form>
      <dl
        class="profile"
      >
        <dt>
          User-ID:
        </dt>
        <dd>
          1234556789
        </dd>
        <dt>
          First name:
        </dt>
        <dd>
          TestUser
        </dd>
      </dl>
      <p>
        Validity: one year (
        1/3/2023
        )
      </p>
      <section
        class="container"
      >
        <div
          class="exclamation"
        />
        <h2
          class="heading"
        >
          Signing error:
        </h2>
        <p
          class="message"
        >
          Your wallet was closed before the request was signed.
        </p>
        <p>
          Click “Continue in Wallet” to try again.
        </p>
      </section>
      <button
        class="ctaButton"
      >
        Continue in Wallet
      </button>
    </form>
  </section>
</div>
`;

exports[`TelegramTemplate should match snapshot with flowError=unknown 1`] = `
<div>
  <section
    aria-busy="false"
    class="container"
  >
    <h1
      class="heading"
    >
      Telegram Account Attestation
    </h1>
    <div
      class="explainerContainer"
    >
      <button
        aria-label="Toggle explanation"
        class="showExplainer"
        type="button"
      />
    </div>
    <section
      class="container"
    >
      <div
        class="exclamation"
      />
      <h2
        class="heading"
      >
        Attestation error:
      </h2>
      <p
        class="message"
      >
        Something went wrong!
      </p>
      <p>
        Click “Try Again” or reload the page or restart your browser.
      </p>
    </section>
    <button
      class="ctaButton"
      type="button"
    >
      Try again
    </button>
    <a
      class="link"
      href="/"
    >
      Back to list
    </a>
  </section>
</div>
`;

exports[`TelegramTemplate should match snapshot with flowerror=unauthorized 1`] = `
<div>
  <section
    aria-busy="false"
    class="container"
  >
    <h1
      class="heading"
    >
      Telegram Account Attestation
    </h1>
    <div
      class="explainerContainer"
    >
      <button
        aria-label="Toggle explanation"
        class="showExplainer"
        type="button"
      />
    </div>
    <section
      class="container"
    >
      <div
        class="exclamation"
      />
      <h2
        class="heading"
      >
        Authorization error:
      </h2>
      <p
        class="message"
      >
        There was an error authorizing your Telegram account.
      </p>
      <p>
        Click “Try Again” or reload the page or restart your browser.
      </p>
    </section>
    <button
      class="ctaButton"
      type="button"
    >
      Try again
    </button>
    <a
      class="link"
      href="/"
    >
      Back to list
    </a>
  </section>
</div>
`;

exports[`TelegramTemplate should match snapshot with status=attesting 1`] = `
<div>
  <section
    aria-busy="false"
    class="container"
  >
    <h1
      class="heading"
    >
      Telegram Account Attestation
    </h1>
    <div
      class="explainerContainer"
    >
      <button
        aria-label="Toggle explanation"
        class="showExplainer"
        type="button"
      />
    </div>
    <section
      class="container"
    >
      <div
        class="spinner"
      />
      <h2
        class="heading"
      >
        Attestation process:
      </h2>
      <p
        class="message"
      >
        Anchoring credential on KILT blockchain
      </p>
      <p>
        Please leave this tab open until your credential is attested.
      </p>
    </section>
  </section>
</div>
`;

exports[`TelegramTemplate should match snapshot with status=authorized 1`] = `
<div>
  <section
    aria-busy="false"
    class="container"
  >
    <h1
      class="heading"
    >
      Telegram Account Attestation
    </h1>
    <div
      class="explainerContainer"
    >
      <button
        aria-label="Toggle explanation"
        class="showExplainer"
        type="button"
      />
    </div>
    <form>
      <dl
        class="profile"
      >
        <dt>
          User-ID:
        </dt>
        <dd>
          1234556789
        </dd>
        <dt>
          First name:
        </dt>
        <dd>
          TestUser
        </dd>
      </dl>
      <p>
        Validity: one year (
        1/3/2023
        )
      </p>
      <button
        class="ctaButton"
      >
        Continue in Wallet
      </button>
    </form>
  </section>
</div>
`;

exports[`TelegramTemplate should match snapshot with status=authorizing 1`] = `
<div>
  <section
    aria-busy="false"
    class="container"
  >
    <h1
      class="heading"
    >
      Telegram Account Attestation
    </h1>
    <div
      class="explainerContainer"
    >
      <button
        aria-label="Toggle explanation"
        class="showExplainer"
        type="button"
      />
    </div>
    <section
      class="container"
    >
      <div
        class="spinner"
      />
      <h2
        class="heading"
      >
        Attestation process:
      </h2>
      <p
        class="message"
      >
        Authorizing
      </p>
      <p>
        Please wait, accessing Telegram account details.
      </p>
    </section>
  </section>
</div>
`;

exports[`TelegramTemplate should match snapshot with status=none 1`] = `
<div>
  <section
    aria-busy="false"
    class="container"
  >
    <h1
      class="heading"
    >
      Telegram Account Attestation
    </h1>
    <div
      class="explainerContainer"
    >
      <button
        aria-label="Toggle explanation"
        class="showExplainer"
        type="button"
      />
    </div>
    <section
      class="container"
    >
      <div
        class="spinner"
      />
      <h2
        class="heading"
      >
        Attestation process:
      </h2>
      <p
        class="message"
      >
        Generating link
      </p>
      <p>
        Please wait, generating Telegram authorization link.
      </p>
    </section>
    <a
      class="link"
      href="/"
    >
      Back to list
    </a>
  </section>
</div>
`;

exports[`TelegramTemplate should match snapshot with status=ready 1`] = `
<div>
  <section
    aria-busy="false"
    class="container"
  >
    <h1
      class="heading"
    >
      Telegram Account Attestation
    </h1>
    <div
      class="explainerContainer"
    >
      <button
        aria-label="Toggle explanation"
        class="showExplainer"
        type="button"
      />
    </div>
    <section
      class="container"
    >
      <div
        class="checkmark"
      />
      <h2
        class="heading"
      >
        Attestation process:
      </h2>
      <p
        class="message"
      >
        Credential is ready
      </p>
      <p>
        We recommend that you back up your credential now.
      </p>
    </section>
    <button
      class="ctaButton"
      type="button"
    >
      Show credential in wallet
    </button>
    <a
      class="link"
      href="/"
    >
      Back to list
    </a>
  </section>
</div>
`;

exports[`TelegramTemplate should match snapshot with status=urlReady 1`] = `
<div>
  <section
    aria-busy="false"
    class="container"
  >
    <h1
      class="heading"
    >
      Telegram Account Attestation
    </h1>
    <div
      class="explainerContainer"
    >
      <button
        aria-label="Toggle explanation"
        class="showExplainer"
        type="button"
      />
    </div>
    <section
      class="container"
    >
      <div
        class="spinner"
      />
      <h2
        class="heading"
      >
        Attestation process:
      </h2>
      <p
        class="message"
      >
        Generating link
      </p>
      <p>
        Please wait, generating Telegram authorization link.
      </p>
    </section>
    <p
      class="buttonsLine"
    >
      <iframe
        class="iframeLoading"
        src="/auth-url"
      />
    </p>
    <a
      class="link"
      href="/"
    >
      Back to list
    </a>
  </section>
</div>
`;

exports[`TelegramTemplate should match snapshot with status=urlReady for real 1`] = `
<div>
  <section
    aria-busy="false"
    class="container"
  >
    <h1
      class="heading"
    >
      Telegram Account Attestation
    </h1>
    <div
      class="explainerContainer"
    >
      <button
        aria-label="Toggle explanation"
        class="showExplainer"
        type="button"
      />
    </div>
    <p
      class="buttonsLine"
    >
      <iframe
        class="iframe"
        src="/auth-url"
      />
    </p>
    <a
      class="link"
      href="/"
    >
      Back to list
    </a>
  </section>
</div>
`;
